### PR TITLE
[consensus] push commit into block_store - refactor 1/n

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

This commit push commit into block_store and simplify sync info
handling.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Existing tests.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

File: consensus/consensus-types/src/sync_info.rs [hunk 1]
```diff
@@ -72,6 +72,10 @@ impl SyncInfo {
             .map_or(0, |tc| tc.round())
     }
 
+    pub fn hcc_round(&self) -> Round {
```
Comment (davidiw) [inline consensus/consensus-types/src/sync_info.rs:75]: why not just call this highest_commit_round?

File: consensus/src/event_processor.rs [hunk 2]
```diff
@@ -396,63 +364,48 @@ impl<T: Payload> EventProcessor<T> {
...
+            debug!(
+                "Local state {} is stale than peer {} remote state {}",
+                local_sync_info,
+                author.short_str(),
+                sync_info
+            );
+            // Some information in SyncInfo is ahead of what we have locally.
+            // First verify the SyncInfo (didn't verify it in the yet).
+            sync_info.verify(&self.epoch_info().verifier).map_err(|e| {
+                security_log(SecurityEvent::InvalidSyncInfoMsg)
```
Comment (davidiw) [inline consensus/src/event_processor.rs:391]: we should probably start stripping out security logs and placing the right libra log / structured log here...

Comment (zekun000) [reply consensus/src/event_processor.rs:391]: yeah, a lot of cleanup work to do

File: consensus/consensus-types/src/sync_info.rs (hunk 1 above)
Comment (zekun000) [reply consensus/consensus-types/src/sync_info.rs:75]: probably lazy? I'll swap them with full name

File: consensus/src/event_processor_test.rs [hunk 3]
```diff
@@ -616,28 +614,35 @@ fn recover_on_restart() {
...
+    let mut data = Vec::new();
     let num_proposals = 100;
     // insert a few successful proposals
-    let a1 = inserter.insert_block_with_qc(certificate_for_genesis(), &genesis, 1);
-    proposals.push(a1);
-    for i in 2..=num_proposals {
-        let parent = proposals.last().unwrap();
-        let proposal = inserter.insert_block(&parent, i, None);
-        proposals.push(proposal);
+    for i in 1..=num_proposals {
```
Comment (davidiw) [inline consensus/src/event_processor_test.rs:623]: first time I noticed this strange syntax

File: consensus/src/block_storage/sync_manager.rs [hunk 4]
```diff
@@ -73,24 +73,49 @@ impl<T: Payload> BlockStore<T> {
...
-            }
-            NeedFetchResult::QCBlockExist => {
-                self.insert_single_quorum_cert(sync_info.highest_quorum_cert().clone())?
-            }
+        self.insert_quorum_cert(sync_info.highest_commit_cert(), &mut retriever)
+            .await?;
+
+        self.insert_quorum_cert(sync_info.highest_quorum_cert(), &mut retriever)
+            .await?;
+
```
[the remark this answers is not in the document]
Comment (davidiw) [reply consensus/src/block_storage/sync_manager.rs:89]: oh gosh, these are different... commit versus quorum, that makes a lot more sense :)

**Pull request closed without merging**